Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibrarySourceProvider.java

```diff
@@ -5,5 +5,5 @@
 import java.io.InputStream;
 
 public interface LibrarySourceProvider {
-    InputStream getLibrarySource(VersionedIdentifier libraryIdentifier);
+    LibraryContentMeta getLibrarySource(VersionedIdentifier libraryIdentifier);
```

Comment:
The implication here is that a LibrarySourceProvider will return all the types of content that it can. IOW, the translator can't request a specific type of content, e.g.:
`InputStream getLibraryContent(VersionedIdentifier libraryIdentifier, ContentType.ELM);`

I think that's not a desirable characteristic. For example, a LIbrarySourceProvider may integrate with an ELM decompiler to produce CQL. But we _only_ want to do that if absolutely necessary because it's expensive to do. Additionally, certain platforms like Android can't support XML even if it's available, so it's a waste to load it. I think we should be able to specify which content to load, and start with the content that we know works best for a given platform, and progressively fall back to CQL.  Alternatively, the LibrarySourceProvider can report the types of content available without loading it (this would be useful for subsetted FHIR Libraries, as an example) and the translator can then make the decision about which to load.

Secondly, this is a breaking change for the public API of the translator. I think it'd be better to implement a second method, and have the existing API remain the same and default to loading CQL.

Reply:
Can the following be a solution?

`public enum LibraryContentType {
    CQL,
    JSON_ELM,
    XML_ELM,
    XML,
    JSON,
    JXSON,
    COFFEE,
    ANY
}`

`public interface LibrarySourceProvider {
    InputStream getLibrarySource(VersionedIdentifier libraryIdentifier);
}`



`public interface LibrarySourceProviderExt extends LibrarySourceProvider {

        boolean isLibrarySourceAvaliable(VersionedIdentifier libraryIdentifier, LibraryContentType type);
	InputStream getLibrarySource(VersionedIdentifier libraryIdentifier, LibraryContentType type);
	
	default InputStream getLibrarySource(VersionedIdentifier libraryIdentifier) {
            return getLibrarySource(libraryIdentifier, LibraryContentType.CQL);
    }
}`

`public interface LibrarySourceLoader {

     void clearProviders();
  
     InputStream getLibrarySource(VersionedIdentifier libraryIdentifier);

     void registerProvider(LibrarySourceProvider provider);
  
}`

`public interface LibrarySourceLoaderExt extends LibrarySourceLoader {

    InputStream getLibrarySource(VersionedIdentifier libraryIdentifier, LibraryContentType type);
    boolean isLibrarySourceAvaliable(VersionedIdentifier libraryIdentifier, LibraryContentType type);

    default InputStream getLibrarySource(VersionedIdentifier libraryIdentifier) {
            return getLibrarySource(libraryIdentifier, LibraryContentType.CQL);
    }
}`

Reply:
I have updated as above not it is backward compatible.

Reply:
Yes this makes sense